### PR TITLE
Strided masked norm.

### PR DESCRIPTION
**Pull request by pearu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #68738
* #68694
* __->__ #68584

Differential Revision: [D32581285](https://our.internmc.facebook.com/intern/diff/D32581285)

### CODE REVIEW

**Comment by cpuhrsch** [comment]
@cpuhrsch has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32581285).

File: test/test_masked.py [hunk 1]
```diff
@@ -12,6 +12,47 @@
     (instantiate_device_type_tests, ops, onlyNativeDeviceTypes)
 
 
+def apply_masked_reduction_along_dim(op, input, *args, **kwargs):
```
Comment (cpuhrsch) [inline test/test_masked.py:15]: Can you add some comments to the code? I find it difficult to follow. In particular the pieces around "dim_position" and ranges.

**Review by cpuhrsch — CHANGES_REQUESTED** [review]
See comment

File: test/test_masked.py (hunk 1 above)
Comment (pearu) [reply test/test_masked.py:16]: Extended the docstring with the description of the algorithm and commented on the code.

**Review by cpuhrsch — APPROVED** [review]

**Comment by cpuhrsch** [comment]
@cpuhrsch has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32581285).

**Pull request closed without merging**